phoenix-core/src/it/java/org/apache/phoenix/end2end/SnapshotTestTemplateIT.java

```diff
@@ -0,0 +1,119 @@
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a not a standard IT.
+ * It is starting point for writing ITs that load specific tables from a snapshot.
+ * Tests based on this IT are meant for debugging specific problems where HBase table snapshots are
+ * available for replication, and are not meant to be part of the standard test suite
+ * (or even being committed to the ASF branches)
+ */
+
+@Ignore
+@Category(NeedsOwnMiniClusterTest.class)
+public class SnapshotTestTemplateIT extends BaseTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(
+        SnapshotTestTemplateIT.class);
+
+    private static final HashMap<String, String> SNAPSHOTS_TO_LOAD;
+
+    static {
+        SNAPSHOTS_TO_LOAD = new HashMap<>();
+        //Add any HBase tables, including Phoenix System tables
+        SNAPSHOTS_TO_LOAD.put("SYSTEM.CATALOG_SNAPSHOT", "/path/to/local/snapshot_dir");
+    }
+
+    @BeforeClass
+    public static synchronized void doSetup() throws Exception {
+        Map<String, String> serverProps = Maps.newHashMapWithExpectedSize(2);
+        serverProps.put(QueryServices.EXTRA_JDBC_ARGUMENTS_ATTRIB, QueryServicesOptions.DEFAULT_EXTRA_JDBC_ARGUMENTS);
+        serverProps.put(QueryServices.IS_NAMESPACE_MAPPING_ENABLED, "true");
+        Map<String, String> clientProps = Maps.newHashMapWithExpectedSize(2);
+        clientProps.put(QueryServices.IS_NAMESPACE_MAPPING_ENABLED, "true");
+
+        //Start minicluster without Phoenix first
+        checkClusterInitialized(new ReadOnlyProps(serverProps.entrySet().iterator()));
+
+        //load snapshots int HBase
+        for (Entry<String, String> snapshot : SNAPSHOTS_TO_LOAD.entrySet()) {
+            importSnapshot(snapshot.getKey(), snapshot.getValue());
+        }
+
+        //Now we can start Phoenix
+        setUpTestDriver(new ReadOnlyProps(serverProps.entrySet().iterator()), new ReadOnlyProps(clientProps.entrySet()
+                .iterator()));
+    }
+
+    private static void importSnapshot(String key, String value) throws IOException {
+        LOGGER.info("importing {} snapshot from {}", key, value);
+        // copy local snapshot dir to Minicluster HDFS
+        Path localPath = new Path(value);
+        assertTrue(FileSystem.getLocal(config).exists(new Path(localPath, ".hbase-snapshot")));
+        FileSystem hdfsFs = FileSystem.get(config);
+        Path hdfsImportPath = new Path(hdfsFs.getHomeDirectory(), "snapshot-import" + "/" + key + "/");
+        assertTrue(hdfsFs.mkdirs(hdfsImportPath));
+        hdfsFs.copyFromLocalFile(localPath, hdfsImportPath);
+        hdfsImportPath = new Path(hdfsImportPath, localPath.getName());
+        assertTrue(hdfsFs.exists(new Path(hdfsImportPath, ".hbase-snapshot")));
+
+        //import the snapshot
+        ExportSnapshot exportTool = new ExportSnapshot();
```

Comment:
Does this automatically run with the `local` mapreduce runner? Not sure where to look to figure out if we already have a MiniMRCluster launched. Either should give you the same results, but debugging embedded MR jobs is painful.